pytest-selenium/pages/content.py

```diff
@@ -4,46 +4,72 @@
 from pages.base import Page
 from regions.base import Region
 
-# TODO: Ensure only 1 <nav> element exists on the Page
-NAV_SELECTOR = "#root > div > div > div > div > nav"
-
 
 class Content(Page):
     URL_TEMPLATE = "/books/{book_slug}/pages/{page_slug}"
 
     _body_locator = (By.TAG_NAME, "body")
     _main_content_locator = (By.CSS_SELECTOR, "h1")
-    _table_of_contents_button_locator = (
-        By.CSS_SELECTOR,
-        "[aria-label='Click to close the Table of Contents'], [aria-label='Click to open the Table of Contents']",
-    )
-    _table_of_contents_nav_locator = (By.CSS_SELECTOR, NAV_SELECTOR)
 
     @property
     def loaded(self):
         return self.find_element(*self._body_locator).get_attribute("data-rex-loaded")
 
     @property
-    def table_of_contents_nav(self):
-        return self.find_element(*self._table_of_contents_nav_locator)
+    def navbar(self):
+        return self.NavBar(self)
 
     @property
-    def table_of_contents_toggle_button(self):
-        return self.find_element(*self._table_of_contents_button_locator)
-
-    def click_table_of_contents_button(self):
-        if self.table_of_contents.is_displayed:
-            self.offscreen_click(self.table_of_contents_toggle_button)
-            return self.wait.until(
-                expected.invisibility_of_element_located(self.table_of_contents_nav)
-            )
-        else:
-            self.table_of_contents_toggle_button.click()
-        return self.table_of_contents.wait_for_region_to_display()
+    def toolbar(self):
+        return self.ToolBar(self)
 
     @property
-    def table_of_contents(self):
-        return self.TableOfContents(self)
+    def sidebar(self):
+        return self.SideBar(self)
+
+    class NavBar(Region):
+        _root_locator = (By.CSS_SELECTOR, "div[class^='NavBar__TopBar']")
```

Comment:
we should just add data selectors

Reply:
Can you show me how to add them? I actually don't think this is that terrible. It works really well and I don't see the prefix portion of these elements changing but I could be wrong.

Reply:
The good thing is that the regions are now more specific. I really like that part. This object representation is a lot closer to how the page is really laid out. This is also much better than XPATH.

Reply:
@TomWoodward ping

Reply:
the class names there are auto generated based on the component names, so if we ever rename a component all the ui tests will break. i would rather use the more explicit coupling of the data selectors.

Reply:
Done.